src/_pytest/assertion/rewrite.py

```diff
@@ -1039,3 +1042,17 @@ def try_mkdir(cache_dir):
             return False
         raise
     return True
+
+
+def get_cache_dir(file_path: Path) -> Path:
+    """Returns the cache directory to write .pyc files for the given .py file path"""
+    if sys.version_info >= (3, 8) and sys.pycache_prefix:
+        # given:
+        #   prefix = '/tmp/pycs'
+        #   path = '/home/user/proj/test_app.py'
+        # we want:
+        #   '/tmp/pycs/home/user/proj'
+        return Path(sys.pycache_prefix) / Path(*file_path.parts[1:-1])
```

Comment:
this seems broken for windows if I had to guess

Reply:
oh I see, it's using `.parts` -- thought this was slicing strings -- I'll have to look into this more :/